test_util/test_upgrade_vpc.py

```diff
@@ -173,9 +172,19 @@ def create_marathon_dns_app(app_id: str, healthcheck_app_id: str) -> dict:
     # DNS resolution app to make sure DNS is available during the upgrade.
     # Periodically resolves the healthcheck app's domain name and logs whether
     # it succeeded to a file in the Mesos sandbox.
+
+    dns_parsed_app_name = marathon_app_id_to_mesos_dns_subdomain(healthcheck_app_id)
+    resolve_name = '{}.marathon.agentip.dcos.thisdcos.directory'.format(dns_parsed_app_name)
+
     return {
         "id": '/' + app_id,
         "cmd": """
+until host -W $TIMEOUT_SECONDS $RESOLVE_NAME
+do
+    echo "Waiting to resolve $RESOLVE_NAME for the first time..."
+    sleep 1
+done
```

Comment:
Thinking about it some more -- we're losing the guarantee that the name was resolvable _before_ the upgrade. If the name isn't resolvable for some reason until after the upgrade, the test will still pass, even though it's not covering what we expect. So if the app itself is responsible for waiting, we have to do something gross like record the time when we start the upgrade and then compare it with the first timestamp in the log. :(

Is there any way for us to maintain that guarantee without comparing timestamps recorded on different machines?